src/state/marketSlice.ts

```
@@ -25,4 +89,25 @@ export const createMarketSlice: StoreSlice<MarketSlice> = (set, get) => ({
 		const market = await getMarket(client, marketPlace);
 		set({ market });
 	},
+	fetchDeals: async (client, marketPlace) => {
+		set({ isLoadingDeals: true });
+		const { allDeals, activeDeals, endedDeals, pendingDeals } = await getDeals(
+			client,
+			marketPlace,
+			get
+		);
+		set({ allDeals, activeDeals, endedDeals, pendingDeals, isLoadingDeals: false });
+	},
+	maybeFetchDeals: async (client, marketPlace) => {
+		await maybeFetchDeals(client, marketPlace, get, set);
+	},
+	getDeal: async (client, marketPlace, dealAddress) => {
+		await maybeFetchDeals(client, marketPlace, get, set);
+
+		if (!get().allDeals) {
+			return;
```

do we expect the deal to exist here? if so I would fail hard. I don't like silent failures because it hides bugs